account_asset_disposal/models/account_asset_asset.py

```diff
@@ -111,6 +50,7 @@ def action_disposal_undo(self):
                 asset.state = 'close'
             else:
                 asset.state = 'open'
+                asset.method_number = asset.category_id.method_number
         return self.write({
```

Comment:
This way auto-recalculates the depreciation board according to the number of entries set in the asset category.